.travis/push_tag

```diff
@@ -1,13 +1,64 @@
 #!/bin/bash
 
-echo "Configure Git and PUSH A TAG - this trigger and execute the deploy section of this same build file, in the same travis linux container (skip_cleanup setting below is important)"
-git config --global user.email "tripleabuilderbot@gmail.com"
-git config --global user.name "tripleabuilderbot"
+#
+# This script pushes a tag to github, that creates a github release and triggers a travis release build
+# which then builds the release artifacts and deploys them to the github release.
+#
+# Note, the tag itself is also considered a branch and trigger an extraneous branch build in travis.
+#
 
-echo "Delete previous tag if it exists, this way we can restart builds and regenerate artifacts in case anything goes wrong"
-git tag -d "$TAGGED_VERSION"
-git push origin ":refs/tags/$TAGGED_VERSION"
+set -u
 
-echo "Push a tag to trigger github releases deployment"
-git tag $TAGGED_VERSION -a -m "$TAGGED_VERSION"
-git push -q https://$GITHUB_PERSONAL_ACCESS_TOKEN_FOR_TRAVIS@github.com/triplea-game/triplea --tags 2>&1 | sed 's|https://.*github|https://[secure]@github|'
+main() {
+  configGithub
+  deleteExistingTag
+  createTag
+  pushTag
+}
+
+configGithub() {
+  echo "Configure Git and PUSH A TAG - this trigger and execute the deploy section of this same build file, in the same travis linux container (skip_cleanup setting below is important)"
+  git config --global user.email "tripleabuilderbot@gmail.com"
+  git config --global user.name "tripleabuilderbot"
+}
+
+
+deleteExistingTag() {
+  echo "Attempting to delete tag $TAGGED_VERSION, this is so we will recreate the release if this build is being re-run."
+  ## sed is to mask out access token in case the https url is printed 
+  git push -q https://$GITHUB_PERSONAL_ACCESS_TOKEN_FOR_TRAVIS@github.com/triplea-game/triplea ":refs/tags/$TAGGED_VERSION" \
+     | sed 's|https://.*github|https://[secure]@github|'
+  git tag -d "$TAGGED_VERSION"
+
+  echo "Done doing tag deletes, ignore any errors immediately above if there are any for failing to delete non-existent tags"
+} 
+
+
+
+createTag() {
+  echo -n "Tagging current branch .. "
+  git tag "$TAGGED_VERSION" -a -m "$TAGGED_VERSION"
```

Comment:
From the [git documentation](https://git-scm.com/docs/git-tag): "Unless `-f` is given the given tag must not exist."
We should probably use this flag instead of manually deleting duplicate old tags...